Source/WebCore/platform/graphics/cairo/GraphicsContextGLCairo.cpp

```diff
@@ -128,7 +128,9 @@ RefPtr<NativeImage> GraphicsContextGL::createNativeImageFromPixelBuffer(const Gr
     RefPtr<cairo_surface_t> imageSurface = adoptRef(cairo_image_surface_create_for_data(
         pixelBuffer->bytes(), CAIRO_FORMAT_ARGB32, imageSize.width(), imageSize.height(), imageSize.width() * 4));
 
-    return NativeImage::create(WTFMove(imageSurface));
+    auto image = NativeImage::create(WTFMove(imageSurface));
```

Comment:
I think it would be more maintainable as a  NativeImage constuctor, into which the PixelBuffer would be moved into.

```
auto image = NativeImage::create(WTFMove(pixelBuffer));
```

Alternatively, Cairo could implement `withDrawingBufferAsNativeImage`, reading directly into a Cairo surface that owns the data, and not going through the PixelBuffer altogether. That would be the best.

Reply:
Changing `withDrawingBufferAsNativeImage` is too big to fix a crash bug. I will take the ctor approach.

Reply:
Fixed.